src/framework/accessibility/internal/accessibilitycontroller.cpp

```diff
@@ -105,6 +105,40 @@ void AccessibilityController::init()
 
     QAccessible::installRootObjectHandler(nullptr);
     QAccessible::setRootObject(s_rootObject);
+
+    if (auto dispatcher = actionsDispatcher()) {
+        dispatcher->preDispatch().onReceive(this, [this](const actions::ActionCode*) {
+            // About to perform an action. Let's store some values to see if the action changes them.
+            m_preDispatchFocus = m_lastFocused;
+            m_preDispatchName = m_preDispatchFocus ? m_preDispatchFocus->accessibleName() : QString();
+            m_announcement.clear(); // So we can detect if the action sets its own announcement.
```

Comment:
@igorkorsukov, are channel `onReceive` functions always called synchronously, or is there a risk that the action was already performed by the time this code executes?

The idea is to store some values before the action is performed, then check them afterwards to see if they were altered by the action.